lib/hydra/works/services/generic_file/add_file_to_generic_file.rb

```diff
@@ -48,21 +48,23 @@ def persist
           end
         end
 
-        def attach_attributes(file) 
+        def attach_attributes(file, mime_type) 
           current_file.content = file
           current_file.original_name = determine_original_name(file)
-          current_file.mime_type = determine_mime_type(file)
+          current_file.mime_type = determine_mime_type(file, mime_type)
         end
 
         # Return mime_type based on methods available to file
         # @param object for mimetype to be determined. Attempts to use methods: :mime_type, :content_type, and :path.
-        def determine_mime_type(file)
+        def determine_mime_type(file, mime_type)
           if file.respond_to? :mime_type
             return file.mime_type
           elsif file.respond_to? :content_type
             return file.content_type
           elsif file.respond_to? :path
             return Hydra::PCDM::GetMimeTypeForFile.call(file.path)
+          elsif mime_type
```

Comment:
If the mime_type is provided shouldn't it have precedence over the result from `GetMimeTypeForFile`, that's basically a wild ass guess based on file extension.